Source/Csla.AspNetCore/ApplicationContextManagerHttpContext.cs

```diff
@@ -0,0 +1,126 @@
+
+namespace Csla.AspNetCore
+{
+  /// <summary>
+  /// Application context manager that uses HttpContextAccessor when 
+  /// resolving HttpContext to store context values.
+  /// </summary>
+  public class ApplicationContextManagerHttpContext : IContextManager
+  {
+
+    private const string _localContextName = "Csla.LocalContext";
+    private const string _clientContextName = "Csla.ClientContext";
+
+    /// <summary>
+    /// Creates an instance of the object, initializing it
+    /// with the required IServiceProvider.
+    /// </summary>
+    /// <param name="httpContextAccessor">HttpContext accessor</param>
+    public ApplicationContextManagerHttpContext(IHttpContextAccessor httpContextAccessor)
+    {
+      HttpContext = httpContextAccessor.HttpContext;
+    }
+
+    /// <summary>
+    /// Gets the current HttpContext instance.
+    /// </summary>
+    protected virtual HttpContext HttpContext { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether this
+    /// context manager is valid for use in
+    /// the current environment.
+    /// </summary>
+    public bool IsValid
+    {
+      get { return HttpContext != null; }
+    }
+
+    /// <summary>
+    /// Gets the current principal.
+    /// </summary>
+    public System.Security.Principal.IPrincipal GetUser()
+    {
+      var result = HttpContext?.User;
+      if (result == null)
+      {
+        result = new Csla.Security.CslaClaimsPrincipal();
+        SetUser(result);
+      }
+      return result;
+    }
+
+    /// <summary>
+    /// Sets the current principal.
+    /// </summary>
+    /// <param name="principal">Principal object.</param>
+    public void SetUser(System.Security.Principal.IPrincipal principal)
+    {
+      HttpContext.User = (ClaimsPrincipal)principal;
+    }
+
+    /// <summary>
+    /// Gets the local context.
+    /// </summary>
+    public ContextDictionary GetLocalContext()
+    {
+      return (ContextDictionary)HttpContext?.Items[_localContextName];
+    }
+
+    /// <summary>
+    /// Sets the local context.
+    /// </summary>
+    /// <param name="localContext">Local context.</param>
+    public void SetLocalContext(ContextDictionary localContext)
+    {
+      HttpContext.Items[_localContextName] = localContext;
+    }
+
+    /// <summary>
+    /// Gets the client context.
+    /// </summary>
+    /// <param name="executionLocation"></param>
+    public ContextDictionary GetClientContext(ApplicationContext.ExecutionLocations executionLocation)
+    {
+      return (ContextDictionary)HttpContext?.Items[_clientContextName];
+    }
+
+    /// <summary>
+    /// Sets the client context.
+    /// </summary>
+    /// <param name="clientContext">Client context.</param>
+    /// <param name="executionLocation"></param>
+    public void SetClientContext(ContextDictionary clientContext, ApplicationContext.ExecutionLocations executionLocation)
+    {
+      HttpContext.Items[_clientContextName] = clientContext;
```

Comment:
No null handling here again, should there be? I'm starting to think that you have ignored the possibility on purpose.

Reply:
We have to throw some exception - nullreference seems valid?